### ISSUE TITLE
Example code to create a machine translation using trax transformer model.

### ISSUE THREAD

**Issue by nag0811** [google/trax#836]
### Description
Hi,
    I am little new to this area. Please excuse me if I am asking any stupid question. I wanted to have machine translation model using Transformers and the model is trained on my own data. Can I get an example code where I can pass my own data as input (both input and target) to the trax transofrmers so that it can predict the translated value. 

I will be using my own laptop for preparing and running the model. It does not have GPU.

Also if possible can I get an example code for speech recognition as well using the transformer model.

Thanks
Nagaraju

...

### Environment information

```
OS: <win10>

$ pip freeze | grep trax
# your output here

$ pip freeze | grep tensor
# your output here

$ pip freeze | grep jax
# your output here

$ python -V
# your output here
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```

**Comment by lukaszkaiser** [comment]
It's a very good idea - we'll try to add a colab soon doing just that!

**Comment by nag0811** [comment]
Thank you 😊

**Comment by MANIKTANEJA3** [comment]
@lukaszkaiser I am currently working on neural machine translation using LSTM and attention mechanism; mind if i contribute to the colab? 
@nag0811 will have an example to work with and it will be helpful for my own project.
Please let me know at your convenience.

**Comment by nag0811** [comment]
> @lukaszkaiser I am currently working on neural machine translation using LSTM and attention mechanism; mind if i contribute to the colab?
> @nag0811 will have an example to work with and it will be helpful for my own project.
> Please let me know at your convenience.

@MANIKTANEJA3 Thanks for the offer. I am working on the data similar to the data set present in http://www.manythings.org/anki/ (for example you can take English to German translation where translation is separated by tab). 
I am looking to get it done using **transformer** model with **python**.

**Comment by MANIKTANEJA3** [comment]
I have used the dataset from this very site. Should you want I can send across the code for the preprocessing. Although, I've used the dataset of English-Hindi for my internship.

**Comment by nag0811** [comment]
> I have used the dataset from this very site. Should you want I can send across the code for the preprocessing. Although, I've used the dataset of English-Hindi for my internship.

Yes, please. What was the accuracy? Please share the code. I can run the same on my own data.

**Comment by MANIKTANEJA3** [comment]
It's the preprocessing sir, but I can complete the code in the next 2 days. I'll attach the google drive link here. Should you want I'll work on the English to German dataset and provide you the complete code. 
https://colab.research.google.com/drive/1EsJ600pojtNbKwjS26WtUEViUWGj4g2R?usp=sharing
this link contains the data preprocessing, model and encoder and attention mechanism; I still need to work correctly with the dimensions so that I can have the correct output for the decoder.

**Comment by nag0811** [comment]
> It's the preprocessing sir, but I can complete the code in the next 2 days. I'll attach the google drive link here. Should you want I'll work on the English to German dataset and provide you the complete code.
> https://colab.research.google.com/drive/1EsJ600pojtNbKwjS26WtUEViUWGj4g2R?usp=sharing
> this link contains the data preprocessing, model and encoder and attention mechanism; I still need to work correctly with the dimensions so that I can have the correct output for the decoder.

@MANIKTANEJA3 Thank you :) I would wait for the complete code. Kindly let me know once it is done. Once again thank you very much. By the way are you using this Trax in your code?

**Comment by MANIKTANEJA3** [comment]
I haven't used Trax, but I am quite hopeful that the preprocessing remains the same; so I can now work with Trax. I'll try to run the scripts and see if I get a result. I'll update you.

**Comment by nag0811** [comment]
> I haven't used Trax, but I am quite hopeful that the preprocessing remains the same; so I can now work with Trax. I'll try to run the scripts and see if I get a result. I'll update you.

I am fine if we use trax or tensor2tensor. Only thing is, if we want to do the model on our own data, we need to create a new "Problem" with our data set and register it. So, it is not something straight forward :(

Also, can I request if you have speechrecognition model as well (I want to convert speech (commands) to text).

**Comment by lukaszkaiser** [comment]
An example of English-German translation (and a pre-trained model) is now in the main Trax README and colab. There is also an explanation of the input processing - I hope this will make it easy for you to create a notebook of your own!

**Comment by nag0811** [comment]
@lukaszkaiser Thanks for the response 👍 
Unfortunately I am not able to run the code on my windows laptop. 
I tried to install trax using pip install trax. But it failed at jaxlib with following error. 
ERROR: Could not find a version that satisfies the requirement tensorflow-text (from trax) (from versions: none)
ERROR: No matching distribution found for tensorflow-text (from trax)

In the notebook, I could see trax is listed as module but when I run the code it is giving "No module named trax"
